src/coreclr/vm/comsynchronizable.cpp

```diff
@@ -686,12 +576,40 @@ FCIMPL1(INT32, ThreadNative::GetApartmentState, ThreadBaseObject* pThisUNSAFE)
     }
 #endif // FEATURE_COMINTEROP
 
-    HELPER_METHOD_FRAME_END();
-
+    END_QCALL;
     return retVal;
 }
-FCIMPLEND
 
+// Indicate whether the thread will host an STA (this may fail if the thread has
+// already been made part of the MTA, use GetApartmentState or the return state
+// from this routine to check for this).
+extern "C" INT32 QCALLTYPE ThreadNative_SetApartmentState(QCall::ThreadHandle thread, INT32 iState)
```

Comment:
Should this take the managed thread as well to maintain the status quo and be consistent with GetApartmentState?

Reply:
Sure.